lib/Sema/CSGen.cpp

```diff
@@ -1345,8 +1345,10 @@ namespace {
       // We may, alternatively, want to use a type variable in that case,
       // and possibly infer the type of the variable that way.
       CS.getTypeChecker().validateDecl(E->getDecl());
-      if (E->getDecl()->isInvalid())
+      if (E->getDecl()->isInvalid()) {
+        E->setType(E->getDecl()->getInterfaceType());
         return nullptr;
+      }
```

Comment:
Yes, I just mentioned that for completeness regarding setting/getting types on Exprs. Your CS.setType call looks correct!